sema/commons/j2/j2_functions.py

```diff
@@ -81,6 +81,15 @@ def xsd_format_date(content, quote, suffix):
         asdt = content
     return xsd_value(asdt.isoformat(), quote, "xsd:date")
 
+def xsd_format_gyear(content, quote, suffix):
+    # make rigid gYear
+    assert (isinstance(content, int) and 1000 <= content <= 9999), "year should be 4-digit integer"
+    return xsd_value(content, quote, "xsd:gYear")
+
+def xsd_format_gmonthyear(content, quote, suffix):
+    # make rigid gMonthYear
+    assert (re.match( r"^--(0[1-9]|1[0-2])-\d{4}$", content) is not None), "month-year should be of format yyyy-mm"
```

Comment:
this regex looks like it is checking `--MM-YYYY` (why the leading -?)  and not `YYYY-MM`